lib/mayaHydra/mayaPlugin/plugin.cpp

```diff
@@ -78,6 +79,44 @@ namespace {
     setenv(name.c_str(), value.c_str(), 1);
     #endif
     }
+
+class SceneModifiedGuard
```

Comment:
Core change in the pull request.  Create an RAII guard class to set the scene as unmodified after loading the mayaHydra plugin.  Plugins often add default nodes on being loaded.  This conceptually does not change the modified state of the scene.